src/main/webapp/components/BuildSnapshot.js

```diff
@@ -1,14 +1,72 @@
-import * as React from "react";
+import * as React from 'react';
+import {getFields} from './utils/getFields.js';
 
 /**
  * Component used to render Build Information.
  *
  * @param {*} props input properties containing build information
  */
 export const BuildSnapshot = (props) => {
+  // State variable indicating whether the BuildSnapshot is open or closed.
+  // isOpen === true indicates the tray should be visible 
+  // and the indicator arrow on the Header should be facing downwards.
+  const [isOpen, setIsOpen] = React.useState(false);
+
+  const headerFields = ['description', 'commitHash', 'repository', 'status'];
+  const trayFields = ['builders', 'timestamp'];
+
+  const headerData = getFields(props.buildData, headerFields, '');
+  const trayData = getFields(props.buildData, trayFields, '');
+
+  return (
+    <div className = 'build-snapshot'>
+      <Header isOpen = {isOpen} onClick = {setIsOpen} data = {headerData}/>
+      <Tray isOpen = {isOpen} data = {trayData}/>
+    </div>
+  );
+}
+
+/**
+  * Subcomponent that holds the content displayed
```

Comment:
The jsdoc comment style has its stars alined with the first star of `/**`, e.g.,
```
/**
 * First line
 *
 * @param ...
 */
```
```

Reply:
Comments re-aligned here 2dd4b10